import os
grouphome = os.environ['GROUPHOME']
ISOLATES = [i for i in open('/home/tjang8858/assembly/nanopore_sra_complete_final.txt').read().split('\n') if len(i) > 0]
#uses assembly conda environment
rule All:
    input:
        expand(grouphome + "/data/depot/assembly/sra_nanopore/{isolate}/mummer/out.snps", isolate = ISOLATES),
        expand(grouphome+ "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf", isolate = ISOLATES),
        expand(grouphome+ "/data/depot/assembly/sra_nanopore/variants/{isolate}_norm.vcf",isolate = ISOLATES)
#dnadiff for each isolate:
rule mummer:
    input:
        input1= grouphome + "/data/depot/assembly/sra_nanopore/{isolate}/assembly/dnaapler/{isolate}_reoriented.fasta",
        input2= grouphome + "/data/genomes/H37Rv-NCBI.fasta"
    output:
        output1=grouphome + "/data/depot/assembly/sra_nanopore/{isolate}/mummer/out.snps",
        output2=directory(grouphome + "/data/depot/assembly/sra_nanopore/{isolate}/mummer")
    shell:
       """cd {output.output2} && dnadiff {input.input2} {input.input1}"""

rule snps2vcf:
    input:
        input1= grouphome + "/data/depot/assembly/sra_nanopore/{isolate}/mummer/out.snps",
        input2= grouphome + "/data/genomes/H37Rv-NCBI.fasta"
    output:
        grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf"
    shell:
        """mummer-snps2vcf {input.input1} --reference {input.input2} > {output}"""
rule zip_vcf:
    input:
        grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf"
    output:
        grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf.gz"
    shell:
        """bgzip < {input} > {output} """
rule index_zip_file:
    input:
        grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf.gz"
    output:
        grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf.gz.tbi"
    shell:
        """bcftools index -t {input}"""
rule normalize_vcf:
    input:
        input1=grouphome + "/data/genomes/H37Rv-NCBI.fasta",
        input2=grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf.gz",
        input3=grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}.vcf.gz.tbi"
    output:
        grouphome + "/data/depot/assembly/sra_nanopore/variants/{isolate}_norm.vcf"
    shell:
        """bcftools norm -f {input.input1} {input.input2} > {output}"""


